server/dialect-idms/src/main/antlr4/org/eclipse/lsp/cobol/dialects/idms/IdmsParser.g4

```diff
@@ -1191,3 +1191,20 @@ cobolCompilerDirectivesKeywords
 endClause
     : (DOT_FS | SEMICOLON_FS)
     ;
+// --------- TOLERATE LR STATEMENTS ---------------//
+idmsLRStatements: obtainLRStatement;
+// obtain LR statement toleration
+obtainLRStatement
+    : OBTAIN (FIRST|NEXT)? logicalRecordName
+      (INTO altLogicalRecordLocation)?
+      (WHERE booleanExpression)?
+      (ON pathStatus imperativeStatement)?
```

Comment:
If I remember correctly the `ON pathStatus statement` gets converted to something like `IF LR-STATUS = pathStatus THEN imperativeStatement`.
If we consume the whole statement none of the lsp actions will work there. Instead we could just replace the `ON pathStatus` with `IF pathStatus`.